### ISSUE TITLE
Allow to replace files with other (mock) ones

### ISSUE THREAD

**Issue by DmitrySharabin** [LeaVerou/htest#25]
In [Madata](https://madata.dev/), we have a natively supported [Firebase backend](https://github.com/madatajs/madata/blob/main/backends/firebase/index.js). However, we face an issue when we need to test it (or any other Madata feature depending on this backend, e.g., the [`Backend.from()`](https://github.com/madatajs/madata/blob/38f6203127c257603048a2726c9e1527c3ac2f40/src/backend.js#L304-L336) method) in Node. The Firebase backend [imports the modules it depends on](https://github.com/madatajs/madata/blob/38f6203127c257603048a2726c9e1527c3ac2f40/backends/firebase/index.js#L4-L7) (provided by Google) via URLs with the `https` scheme, which Node's default ESM loader does not support.

It would be nice if hTest could provide a way to replace one file (e.g., by intercepting the corresponding request) with another one on the fly. For example, we could provide hTest a mock Firebase backend (with the functionality we need to test) instead of the real one. This feature will allow hTest to avoid choking over features unsupported by Node (or any other environment the tests run in).

This enhancement might also be helpful in other cases, such as when we need to test features requiring some user actions, like authentication.

As another option, hTest might support something similar to Angular's [Dependency Injection](https://angular.io/guide/dependency-injection-overview), where [the author can use some existing class](https://angular.io/guide/dependency-injection-in-action#class-interface) instead of the injected one. If I'm correct, this feature is heavily used exactly when some class (component, directive, etc.) needs to be tested, and it depends on a (injected) class that is replaced by another (mock) class.

In other words, it would be helpful if hTest supports something like (unsure about the API, just an example):

```js
export default {
	replace: {
		"./src/foo.js": "./test/foo.js",
	},
	tests: [...],
};
```

OR

```js
export default {
	use: {
		"./src/foo.js": "./test/foo.js",
	},
	tests: [...],
};
```
